### PR TITLE
Deploy saver

### PR DESCRIPTION
**Pull request by DevAseel**
**Summary of change:**
- Added a new micro-service, deploy saver, which takes a parsed deploy and save it in the datasource. 

**List of tasks:** 
- [x] Define types.
- [x] Define interfaces.
- [x] Define class.
- [x] Create fixture.
- [x] Create mock datasource. 
- [x] Create unit test.
- [x] Generate the docs. 
- [x] Update the documentation.

**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/deploy-saver`
```bash
git checkout feature/deploy-saver
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm deploy-saver test has passed.
- Run the docs using live server and make sure it have been updated.

### CODE REVIEW

**Comment by jjfluid** [comment]
> 
>     * Run the test across the test-net
> 

The tests only worked on mainnet for me.

File: src/entities/Deploy.ts [hunk 1]
```diff
@@ -1,13 +1,59 @@
...
+  @Column({ name: 'block_number', nullable: false })
+  blockNumber: number;
+
+  @Column({ name: 'cost', type: 'varchar', length: 155 })
+  cost: string;
+
+  @Column({ name: 'payment', type: 'jsonb' })
+  payment: object;
+
+  @Column({ name: 'session', type: 'jsonb' })
```
Comment (jjfluid) [inline src/entities/Deploy.ts:54]: remove this (session) - will contain the huge moduleBytes hex field, which we dont want in our data store

**Comment by DevAseel** [comment]
> > ```
> > * Run the test across the test-net
> > ```
> 
> The tests only worked on mainnet for me.

@jjfluid I'm assuming you mean the deploy fetcher integration test, for that we need to find some samples that exist in both test-net and main-net. 

```js
const validTestCases: string[] = [
      '57a2fc930ab83f6bf5b3f1d7d5d1bd05ccd20dcbe3135f7c21a0553650d220e6',
      '2ed45dff7b6ca5fde781fd79ae9cb4c68c69c63ac4898d205f937ac05cd4e387',
      'da890a18935ddf8c27760add34bde7c2479b96cb3282d95935b1652167b01abf',
    ];
```

I included test samples from the main-net to pass the test.

File: src/services/parsers/DeployParser.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        dependencies: deploy.fields.deploy.header.dependencies,
+        chainName: deploy.fields.deploy.header.chain_name,
+        parsedPaymentAmount: deploy.fields.deploy.payment.ModuleBytes.args[0][1].parsed,
+        contractHash: deploy.fields.deploy.session.StoredContractByHash.hash,
+        entryPoint: deploy.fields.deploy.session.StoredContractByHash.entry_point,
+        blockHash: deploy.fields.execution_results[0].block_hash,
+        blockNumber: deploy.blockNumber,
+        cost: deploy.fields.execution_results[0].result.Success.cost,
+        payment: deploy.fields.deploy.payment,
+        session: deploy.fields.deploy.session,
```
Comment (jjfluid) [inline src/services/parsers/DeployParser.ts:42]: remove this...too big of a field, with data we don't need

**Review by jjfluid — CHANGES_REQUESTED** [review]

**Comment by jjfluid** [comment]
> 

That's right. It was the DeployFetcher int. tests, on this branch.

**Comment by DevAseel** [comment]
> Everything else looks good. Nice work

Thanks, will include the changes [here](https://github.com/fluidefi/fluidefi-caspernet-aggregator-tools/pull/47) if you don't mind, just to avoid conflicts. 

> > 
> 
> That's right. It was the DeployFetcher int. tests, on this branch.

For that we need to find a deploy hash that exists in main-net & test-net.

**Comment by jjfluid** [comment]
> 
> Thanks, will include the changes [here](https://github.com/fluidefi/fluidefi-caspernet-aggregator-tools/pull/47) if you don't mind, just to avoid conflicts.
> 

Yeah sounds good

**Comment by DevAseel** [comment]
@specialistvlad this one should be closed without merging, all changes are included in #[47](https://github.com/fluidefi/fluidefi-caspernet-aggregator-tools/pull/47).

**Comment by specialistvlad** [comment]
@DevAseel ok

**Pull request closed without merging**